localization:
    rate: 50.0
    num_particles: 500
    initial_state:
        x_pos: -30.0
        y_pos: -20.0
        z_pos: 4.9
        x_lin_vel: 0.0
        y_lin_vel: 0.0
        z_lin_vel: 0.0
    variance:
        # measurement covariance matrix
        hydrophone_x_variance: 0.5
        hydrophone_y_variance: 0.5
        hydrophone_z_variance: 0.5
        depth_variance: 0.1

        # state update covariance matrix
        state_update_x_pos_variance: 0.2
        state_update_y_pos_variance: 0.2
        state_update_z_pos_variance: 0.2

        # initial distribution stddevs
        x_pos_initial_stddev: 5.0
        y_pos_initial_stddev: 5.0
        z_pos_initial_stddev: -2.0

kalman_filter:
    Q: 
       - [0.1, 0, 0, 0, 0, 0, 0, 0, 0]
       - [0, 0.1, 0, 0, 0, 0, 0, 0, 0]
       - [0, 0, 0.1, 0, 0, 0, 0, 0, 0]
       - [0, 0, 0, 0.1, 0, 0, 0, 0, 0]
       - [0, 0, 0, 0, 0.1, 0, 0, 0, 0]
       - [0, 0, 0, 0, 0, 0.1, 0, 0, 0]
       - [0, 0, 0, 0, 0, 0, 0.1, 0, 0]
       - [0, 0, 0, 0, 0, 0, 0, 0.1, 0]
       - [0, 0, 0, 0, 0, 0, 0, 0, 0.1]
    R: 
       - [0.4, 0, 0, 0]
       - [0, 0.4, 0, 0]
       - [0, 0, 0.4, 0]
       - [0, 0, 0, 0.4]
    P: 
       - [0.4, 0, 0, 0, 0, 0, 0, 0, 0]
       - [0, 0.4, 0, 0, 0, 0, 0, 0, 0]
       - [0, 0, 0.4, 0, 0, 0, 0, 0, 0]
       - [0, 0, 0, 0.4, 0, 0, 0, 0, 0]
       - [0, 0, 0, 0, 0.4, 0, 0, 0, 0]
       - [0, 0, 0, 0, 0, 0.4, 0, 0, 0]
       - [0, 0, 0, 0, 0, 0, 0.4, 0, 0]
       - [0, 0, 0, 0, 0, 0, 0, 0.4, 0]
       - [0, 0, 0, 0, 0, 0, 0, 0, 0.4]
    X0: 
        - [-30.0]
        - [-20.0]
        - [4.9]
        - [0]
        - [0]
        - [0]
        - [0]
        - [0]
        - [0]
